include/ginkgo/core/matrix/permutation.hpp

```diff
@@ -114,6 +114,23 @@ class Permutation : public EnableLinOp<Permutation<IndexType>>,
         return permutation_.get_num_elems();
     }
 
+    /**
+     * Get the permute masks
+     *
+     * @param permute_mask the permute masks
+     */
+    mask_type get_permute_mask() { return enabled_permute_; }
+
+    /**
+     * Set the permute masks
+     *
+     * @param permute_mask the permute masks
+     */
+    void set_permute_mask(mask_type permute_mask)
```

Comment:
You could actually also make this method const and change `enabled_permute_` to be mutable. Then you are sure this method will never touch the data (you have a core test which checks that, which is good).
That's up to you.

Reply:
I would actually prefer if it stayed the way it is right now since `enabled_permute_` can change the behavior of this class drastically (see the `apply_impl`). If you could change that while it is const (e.g. in a `std::shared_ptr<const Permutation>`), you were allowed to change it anywhere which can lead to unwanted/wrong results without an easy way to track down what went wrong.

Reply:
Right, that makes sense.